CMakeLists.txt

```diff
@@ -1458,6 +1491,15 @@ ocv_build_features_string(parallel_status EXCLUSIVE
   ELSE "none")
 status("")
 status("  Parallel framework:" "${parallel_status}")
+if (OPENCV_DISABLE_THREAD_SUPPORT)
+  status("" "Multi thread code explicitly disabled with OPENCV_DISABLE_THREAD_SUPPORT.")
+  if(HAVE_PTHREADS_PF OR HAVE_HPX OR HAVE_OPENMP OR HAVE_GCD OR HAVE_CONCURRENCY)
+    message(WARNING "Not all parallel frameworks have been disabled (using ${parallel_status}).")
```

Comment:
I propose use FATAL message for the case, because `OPENCV_DISABLE_THREAD_SUPPORT` disables Mutex and other sync primitives and makes parallel implementations unsafe.